nwb_conversion_tools/utils.py

```diff
@@ -48,24 +50,155 @@ def get_schema_from_method_signature(class_method):
     return input_schema
 
 
-def get_schema_from_docval(docval):
+def get_schema_from_hdmf_class(hdmf_class):
+    schema = get_schema_from_docval(hdmf_class.__init__.__docval__)
+    schema['tag'] = hdmf_class.__module__ + '.' + hdmf_class.__name__
+    return schema
+
 
+def get_schema_from_docval(docval):
+    pynwb_link_types = [pynwb.device.Device, pynwb.ophys.ImagingPlane, pynwb.image.ImageSeries]
     schema = get_base_schema()
     for docval_arg in docval['args']:
-        if docval_arg['type'] is str or (isinstance(docval_arg['type'], tuple) and str in docval_arg['type']):
-            schema_arg = dict(name=docval_arg['name'], type='string', description=docval_arg['doc'])
-            if 'default' in docval_arg:
-                if docval_arg['default'] is not None:
-                    schema_arg.update(default=docval_arg['default'])
+        schema_arg = dict()
+
+        # type float
+        if docval_arg['type'] == 'float' or (isinstance(docval_arg['type'], tuple) and 'float' in docval_arg['type']):
+            schema_arg[docval_arg['name']] = dict(type='number', description=docval_arg['doc'])
+
+        # type string
+        elif docval_arg['type'] is str or (isinstance(docval_arg['type'], tuple) and str in docval_arg['type']):
+            schema_arg[docval_arg['name']] = dict(type='string', description=docval_arg['doc'])
+
+        # type datetime
+        elif docval_arg['type'] is datetime or (isinstance(docval_arg['type'], tuple) and datetime in docval_arg['type']):
+            schema_arg[docval_arg['name']] = dict(type='string', description=docval_arg['doc'], format='date-time')
+
```

Comment:
This is good, but is there a way to refactor these lines so that you are repeating less code?